cmd/workspace/secrets/overrides.go

```diff
@@ -71,12 +82,19 @@ var putSecretCmd = &cobra.Command{
 			putSecretReq.Scope = args[0]
 			putSecretReq.Key = args[1]
 
-			value, err := cmdio.Secret(ctx)
-			if err != nil {
-				return err
+			// Turn bytes value into a base64 encoded version of itself if specified.
+			if bytesValueChanged {
+				putSecretReq.BytesValue = base64.StdEncoding.EncodeToString([]byte(putSecretReq.BytesValue))
 			}
 
-			putSecretReq.StringValue = value
+			// Read secret value from stdin if not specified.
+			if !bytesValueChanged && !stringValueChanged {
+				bytes, err := promptSecret(cmd)
+				if err != nil {
+					return err
+				}
+				putSecretReq.BytesValue = base64.StdEncoding.EncodeToString(bytes)
+			}
```

Comment:
Missing a final else case for stringValueChanged != nil?

Reply:
When `string-value` flag is set, `cobra` reads the value directly into `putSecretReq.StringValue` on line 27